test/unit/utility/bloom_filter/bloom_filter_test.cpp

```diff
@@ -0,0 +1,144 @@
+                 std::logic_error);
+}
+
+TYPED_TEST(bloom_filter_test, member_getter)
+{
+    TypeParam t1{TestFixture::make_bf(seqan3::bin_size{1024u})};
+    EXPECT_EQ(t1.size(), 1024u);
+    EXPECT_EQ(t1.hash_function_count(), 2u);
+
+    TypeParam t2{TestFixture::make_bf(seqan3::bin_size{1019u},
+                                       seqan3::hash_function_count{3u})};
+    EXPECT_EQ(t2.size(), 1019u);
+    EXPECT_EQ(t2.hash_function_count(), 3u);
+}
+
+TYPED_TEST(bloom_filter_test, contains)
+{
+    TypeParam bf{TestFixture::make_bf(seqan3::bin_size{1024u})};
+
+    for (size_t hash : std::views::iota(0u, 64u)) // test some hashes
+    {
+        // Expect false for all queries since we did not insert anything
+        EXPECT_FALSE(bf.contains(hash)); 
+    }
+}
+
+TYPED_TEST(bloom_filter_test, counting)
+{
+    // 1. Test uncompressed bloom_filter directly because the compressed one is not mutable.
+    seqan3::bloom_filter bf{seqan3::bin_size{1024u},
+                            seqan3::hash_function_count{2u}};
+
+    for (size_t hash : std::views::iota(0u, 128u))
+        bf.emplace(hash);
+
+    // 2. Construct either the uncompressed or compressed bloom_filter and test set with bulk_contains
+    TypeParam bf2{bf};
+    
+    // Test counting with all elements
+    EXPECT_EQ(bf2.count(std::views::iota(0u, 128u)), 128u);
+
+    // Test counting with some elements
+    EXPECT_EQ(bf2.count(std::views::iota(22u, 42u)), 20u);
+}
+
+TYPED_TEST(bloom_filter_test, emplace)
+{
+    // 1. Test uncompressed bloom_filter directly because the compressed one is not mutable.
+    seqan3::bloom_filter bf{seqan3::bin_size{1024u},
+                            seqan3::hash_function_count{2u}};
+
+    for (size_t hash : std::views::iota(0u, 64u))
+        bf.emplace(hash);
+
+    // 2. Construct either the uncompressed or compressed bloom_filter and test via hit counts
+    TypeParam bf2{bf};
+    EXPECT_EQ(bf.count(std::views::iota(0u, 64u)), 64u); // test inserted hashes
+}
+
+TYPED_TEST(bloom_filter_test, clear)
+{
+    // 1. Test uncompressed bloom_filter directly because the compressed one is not mutable.
+    seqan3::bloom_filter bf{seqan3::bin_size{1024u},
+                            seqan3::hash_function_count{2u}};
+
+    for (size_t hash : std::views::iota(0u, 64u))
+        bf.emplace(hash);
+
+    // 2. Clear the Bloom Filter
+    bf.clear();
+
+    // 3. Construct either the uncompressed or compressed bloom_filter and test set with contains
+    TypeParam bf2{bf};
+    EXPECT_EQ(bf.count(std::views::iota(0u, 64u)), 0u); // nothing should be present in the bloom filter
+}
+
+TYPED_TEST(bloom_filter_test, data_access)
+{
+    seqan3::bloom_filter bf{seqan3::bin_size{1024u}};
+    EXPECT_LE(sdsl::size_in_mega_bytes(bf.raw_data()), 0.001f);
+}
+
+TYPED_TEST(bloom_filter_test, serialisation)
+{
+    TypeParam bf{TestFixture::make_bf(seqan3::bin_size{1024u})};
+    seqan3::test::do_serialisation(bf);
+}
+
```

Comment:
```suggestion
    seqan3::test::do_serialisation(bf);
}
```

I assume this is github's way of telling me "two new lines at end"